### PR TITLE
Updated location Param to Address Object param in Organizations (Updated).

### PR DESCRIPTION
**Pull request by Anubhav-2003**
**What kind of change does this PR introduce?**
A feature.

**Issue Number:**
#1650 
Fixes #

**Did you add tests for your changes?**
Yes

**Snapshots/Videos:**

**If relevant, did you update the documentation?**
No.

**Summary**

**Does this PR introduce a breaking change?**
Yes, this PR is an updated PR for the initially merged PR https://github.com/PalisadoesFoundation/talawa-api/pull/1666. 
This PR must be merged with the corresponding ADMIN portal PR: https://github.com/PalisadoesFoundation/talawa-admin/pull/1490 for successful and complete implementation of the feature.

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

No.

### CODE REVIEW

**Comment by Cioppolo14** [comment]
@xoldyckk @noman2002  Please review this PR, there is also a corresponding PR almost ready in talwa-admin (https://github.com/PalisadoesFoundation/talawa-admin/pull/1490).

File: src/models/Organization.ts [hunk 1]
```diff
@@ -66,8 +74,31 @@ const organizationSchema = new Schema(
       type: String,
       required: true,
     },
-    location: {
-      type: String,
+    address: {
```
Comment (xoldyckk) [inline src/models/Organization.ts:77]: confirm which fields are required and add `required: true` to them

Comment (Anubhav-2003) [reply src/models/Organization.ts:77]: Ok sir, I will make those two mandatory. As for the enums, I have already hardcoded an entire list of objects with country codes and country names in the admin portal for the drop down list. I found that creating an object would be more self-explanatory and better as compared to just an enum. you can find that change in the other PR. Thank you.

Comment (Anubhav-2003) [reply src/models/Organization.ts:77]: @palisadoes @xoldyckk Respected sir, I have made the changes requested, and made countrycode, and city mandatory. But due to this change. Some tests are failing. I will be fixing those tests as early as possible. 

Thank You.

**Comment by Anubhav-2003** [comment]
@palisadoes @xoldyckk Respected sir, I have completely implemented the feature both in the API, and the ADMIN, and it is working properly. I have also made the countryCode, and city in the address mandatory from both the ADMIN portal, and the  createOrganization resolver. The ADMIN PR has passed all the tests. 

But one major problem for making the address countrycode, and city mandatory from the schema level is the fact that currently the location param is not mandatory in the schema. As a result most of the test cases have been created with their own instances of the organization without the location param (and without a single point of truth),  now making the address mandatory at that level would require substantial changes to all the test files. Now I can make all the changes but it would take time, as every test file has its own logic which I would need to examine before making the changes.

This would lead to unnecessary delays to this feature that is completely functional.  As the fields have already been mandated at the resolver level, as well as from the frontend. 

Thank You.

**Comment by palisadoes** [comment]
1. The issue stated:
    1. `Replace the location field with the existing address type`
1. Location is deprecated and must not be supported

**Comment by Anubhav-2003** [comment]
@palisadoes ok sir, I will update the setup.ts. And most of the work is done. I will request for review as soon as everything is finalised.

Thank you.

**Comment by Anubhav-2003** [comment]
@palisadoes Respected sir, I have already made the required changes in the sample data, and the sampleOrganization create resolver to accommodate for the changes in the fields. 

The updated data is already part of this PR. I have checked, and the sample data is being imported correctly. I am attaching a screenshot of it too.

Thank You.
Screenshot from 2024-01-23 14-18-00

**Comment by Anubhav-2003** [comment]
> 1. The issue stated:
>        
>        1. `Replace the location field with the existing address type`
> 
>     2. Location is deprecated and must not be supported
> 
>     3. The sample data as part of the `setup.ts` script needs to be updated  to support this new requirement
> 
>     4. Issues have been created in other repos to support this PR

Sir, you have created a duplicate issue for this PR in the ADMIN portal. I have already been assigned, and have already Raised a PR: https://github.com/PalisadoesFoundation/talawa-admin/pull/1490 for it in the ADMIN portal

Thank You.

**Review by palisadoes — APPROVED** [review]

**Comment by SiddheshKukade** [comment]
@Anubhav-2003 just wanted to clarify are there any updations required in `setup.ts` for this? If yes have you implemented them or not? Please reply.

**Comment by Anubhav-2003** [comment]
@SiddheshKukade Respected sir, 

Since my change mostly dealt with change in the structure of Organization type. So in terms of initial setup, I had to change the structure of the Sample Organizations, and the utility function that created the sample orgs. No changes were required in my opinion in the setup.ts. I checked, and the updated sample organizations are being imported correctly into the DB after the PR.

Thank You.

**Comment by Dante291** [comment]
@Anubhav-2003 I have been assigned to fix this issue in mobile repo, I need some help of yours to fix this but I am unable to find you on slack could you please drop a text to me on slack(same name and pfp) Thanks.

**Pull request merged**